falcon_kit/mains/ovlp_to_graph.py

```diff
@@ -1146,55 +1197,75 @@ def main(argv=sys.argv):
-    utg_spurs = set()
     all_nodes = ug.nodes()
 
     ug2 = ug.copy()
-    spur_edges = set()
-    edges_to_remove = set()
 
-    for n in s_nodes:
-        if ug.in_degree(n) != 0:
+    s_candidates = set( )
+    for v in ug2.nodes():
+        if ug2.in_degree(v) == 0:
+            s_candidates.add(v)
+
+    while len(s_candidates) > 0:
+        n = s_candidates.pop()
+        if ug2.in_degree(n) != 0:
             continue
-        for s, t, v in ug.out_edges(n, keys=True):
-            length, score, edges, type_ = u_edge_data[ (s, t, v) ]
-            if length > 50000 and len(edges) > 3:
+        n_ego_graph = nx.ego_graph( ug2, n, radius = 10 )
+        n_egg_node_set = set( n_ego_graph.nodes() )
+        for b_node in n_ego_graph.nodes():
+            if ug2.in_degree(b_node) <= 1:
                 continue
-            in_degree = len( set( e[0] for e in ug.in_edges(t))  ) # ignore mutli-edges
-            out_degree = len( set( e[1] for e in ug.out_edges(t)) )
-            if in_degree > 1 and out_degree > 0:
-                spur_edges.add( (s, t, v) )
-                edges_to_remove.add( (s, t, v) )
-                u_edge_data[ (s, t, v) ] = length, score, edges, "spur:2"
-                rs = reverse_end(t)
-                rt = reverse_end(s)
-                rv = reverse_end(v)
-                edges_to_remove.add( (rs, rt, rv) )
-                length, score, edges, type_ = u_edge_data[ (rs, rt, rv) ]
-                u_edge_data[ (rs, rt, rv) ] = length, score, edges, "spur:2"
 
-    for n in t_nodes:
-        if ug.out_degree(n) != 0:
-            continue
-        for s, t, v in ug.in_edges(n, keys=True):
-            length, score, edges, type_ = u_edge_data[ (s, t, v) ]
-            if length > 50000 and len(edges) > 3:
+            with_extern_node = False
+            b_in_nodes = [e[0] for e in ug2.in_edges(b_node)]
+
+            if len(b_in_nodes) == 1:
+                continue
+
+            for v in  b_in_nodes:
+                if v not in n_egg_node_set:
+                    with_extern_node = True
+                    break
+
+            if not with_extern_node:
                 continue
-            in_degree = len( set( e[0] for e in ug.in_edges(s))  ) # ignore mutli-edges
-            out_degree = len( set( e[1] for e in ug.out_edges(s)) )
-            if in_degree > 0 and out_degree > 1:
-                spur_edges.add( (s, t, v) )
-                edges_to_remove.add( (s, t, v) )
-                u_edge_data[ (s, t, v) ] = length, score, edges, "spur:2"
-                rs = reverse_end(t)
-                rt = reverse_end(s)
-                rv = reverse_end(v)
-                edges_to_remove.add( (rs, rt, rv) )
-                length, score, edges, type_ = u_edge_data[ (rs, rt, rv) ]
-                u_edge_data[ (rs, rt, rv) ] = length, score, edges, "spur:2"
 
-    for s, t, v in list(edges_to_remove):
-        ug2.remove_edge( s, t, key= v)
+            s_path = nx.shortest_path( ug2, n, b_node )
```

Comment:
This line is unreachable. If `with_extern_node` is false, then we `continue` at 1231; if it's true, then we `break` at 1228.